sdk/deepbook-v3/examples/deepbookMarketMaker.ts

```diff
@@ -19,24 +18,32 @@ export class DeepBookMarketMaker {
 		env: 'testnet' | 'mainnet',
 		balanceManagers?: { [key: string]: BalanceManager },
 	) {
+		let resolvedKeypair: Keypair;
+
 		if (typeof keypair === 'string') {
-			keypair = this.#getSignerFromPK(keypair);
+			resolvedKeypair = DeepBookMarketMaker.getSignerFromPK(keypair);
+		} else {
+			resolvedKeypair = keypair;
 		}
-		this.keypair = keypair;
-		const suiClient = new SuiClient({
-			url: getFullnodeUrl(env),
-		});
-		const address = keypair.toSuiAddress();
-		this.dbClient = new DeepBookClient({
+
+		const address = resolvedKeypair.toSuiAddress();
+
+		super({
 			address: address,
 			env: env,
-			client: suiClient,
+			client: new SuiClient({
+				url: getFullnodeUrl(env),
+			}),
 			balanceManagers: balanceManagers,
 		});
-		this.suiClient = suiClient;
+
+		this.keypair = resolvedKeypair;
+		this.suiClient = new SuiClient({
+			url: getFullnodeUrl(env),
+		});
 	}
 
-	#getSignerFromPK = (privateKey: string) => {
+	static getSignerFromPK = (privateKey: string) => {
```

Comment:
I am guessing you made this static because you realized you can't actually use the client without also getting the keypair.

The client lets you construct transactions, but without the keypair, you can't execute them.  I adding an executeTransaction method back to this class so that you can do something like:


```ts
const mmClient = new DeepBookMarketMaker(privateKey, 'testnet')

const tx = new Transaction();
mmClient.placeLimitOrderExample(tx);
mmClient.flashLoanExample(tx);

await mmClient.executeTransaction(tx);
```

Reply:
@hayes-mysten Let me know how this would look different compared to the signAndExecute(tx) function currently